driver-core/src/main/com/mongodb/connection/netty/NettyStream.java

```diff
@@ -174,6 +174,7 @@ public void writeAsync(final List<ByteBuf> buffers, final AsyncCompletionHandler
             io.netty.buffer.ByteBuf byteBuf = ((NettyByteBuf) cur).asByteBuf();
             composite.addComponent(byteBuf.retain());
             composite.writerIndex(composite.writerIndex() + byteBuf.writerIndex());
+            byteBuf.release();
```

Comment:
Looking a bit more closely, I'm thinking there should be a release of 'composite' after writeAndFlush completes.